src/cpu/pred/BranchPredictor.py

```diff
@@ -63,14 +178,23 @@ class BranchPredictor(SimObject):
     abstract = True
 
     numThreads = Param.Unsigned(Parent.numThreads, "Number of threads")
-    BTBEntries = Param.Unsigned(4096, "Number of BTB entries")
-    BTBTagSize = Param.Unsigned(16, "Size of the BTB tags, in bits")
-    RASSize = Param.Unsigned(16, "RAS size")
     instShiftAmt = Param.Unsigned(2, "Number of bits to shift instructions by")
+    requiresBTBHit = Param.Bool(
+        False,
+        "Requires a BTB hit to detect if "
+        " a branch was a return or indirect branch.",
+    )
 
+    RASSize = Param.Unsigned(16, "RAS size")
+
+    BTB = Param.BranchTargetBuffer(SimpleBTB(), "Branch target buffer (BTB)")
```

Comment:
I'd suggest calling these `btb` and `ras` instead since we don't normally capitalise abbreviations on their own (see for example `mmu` in `BaseCPU`). It's likely that there will be modules with CPU configurations containing a BTB class called just `BTB`, so this would avoid a bit of confusion.